### PR TITLE
Add pagination to list_witnesses and list_proposals

### LINKED ISSUE #21: Add pagination to list_witnesses and list_proposals
## Summary

`list_witnesses` and `list_proposals` return full lists with no pagination, causing large responses that can exceed MCP token limits.

## Current Behavior

- `list_witnesses` returns all ~27+ SRs in one response
- `list_proposals` returns all proposals (can be hundreds)
- No `limit` or `offset` parameters

## Desired Behavior

Add `limit` and `offset` parameters matching the pattern already used in `get_block`:

```json
{
  "limit": 10,
  "offset": 0
}
```

Response includes:
- `total` — total count
- `returned` — count in this page
- `has_more` — boolean
- `next_offset` — for next page

## Files

- `internal/tools/witness.go` — `handleListWitnesses`
- `internal/tools/proposal.go` — `handleListProposals`

## Reference

- `internal/tools/block.go` — existing pagination pattern (lines 84-111)

### PR DESCRIPTION
**Pull request by fbsobreira**
## Summary

- Add `limit` and `offset` pagination parameters to `list_witnesses` and `list_proposals` tools
- Response now includes `total`, `returned`, `has_more`, and `next_offset` metadata
- Add `order` parameter to `list_proposals` (`desc` default = newest first, `asc` = oldest first)
- Add mock gRPC server tests for both tools covering pagination edge cases, ordering, and empty lists

Closes #21

## Test plan

- [x] `make test` — all tests pass with race detector
- [x] `make lint` — 0 issues
- [x] `make build` — builds successfully

## Summary by CodeRabbit

* **New Features**
  * Added pagination to list_proposals (limit, offset, order) and list_witnesses (limit, offset); responses now return paged items plus total/returned and optional has_more/next_offset.
* **Bug Fixes**
  * Validates sort order and normalizes negative/invalid limit/offset values.
* **Tests**
  * Added extensive unit tests covering pagination, ordering, edge cases, and JSON response shape for both proposals and witnesses.

### CODE REVIEW

File: internal/tools/witness.go [hunk 1]
```diff
@@ -36,6 +38,9 @@ func RegisterWitnessWriteTools(s *server.MCPServer, pool *nodepool.Pool) {
 
 func handleListWitnesses(pool *nodepool.Pool) server.ToolHandlerFunc {
 	return func(ctx context.Context, req mcp.CallToolRequest) (*mcp.CallToolResult, error) {
+		limit := req.GetInt("limit", 10)
+		offset := req.GetInt("offset", 0)
```
[the remark this answers is not in the document]
Comment (fbsobreira) [reply internal/tools/witness.go]: Fixed in 03e24ab — added validation to clamp negative limit to default 10 and negative offset to 0. Also added a test for this edge case.
